src/daft-core/src/series/ops/list.rs

```diff
@@ -254,4 +257,76 @@ impl Series {
     pub fn list_fill(&self, num: &Int64Array) -> DaftResult<Self> {
         ListArray::list_fill(self, num).map(|arr| arr.into_series())
     }
+
+    /// Returns a list of unique elements in each list, preserving order of first occurrence and ignoring nulls.
+    ///
+    /// # Example
+    /// ```txt
+    /// [[1, 2, 3], [1, 1, 1], [NULL, NULL, 5]] -> [[1, 2, 3], [1], [5]]
+    /// ```
+    pub fn list_distinct(&self) -> DaftResult<Self> {
+        let input = if let DataType::FixedSizeList(inner_type, _) = self.data_type() {
+            self.cast(&DataType::List(inner_type.clone()))?
+        } else {
+            self.clone()
+        };
+
+        let list = input.list()?;
+        let mut offsets = Vec::new();
+        offsets.push(0i64);
+        let mut current_offset = 0i64;
+
+        let field = Arc::new(input.field().to_exploded_field()?);
+        let child_data_type = if let DataType::List(inner_type) = input.data_type() {
+            inner_type.as_ref().clone()
+        } else {
+            return Err(DaftError::TypeError("Expected list type".into()));
+        };
+
+        // Create growable with the flat child as source, overestimating capacity
+        let mut growable = make_growable(
+            &field.name,
+            &child_data_type,
+            vec![&list.flat_child],
+            false,
+            list.flat_child.len(),
+        );
+
+        // Single pass: process each sub-series
+        let list_offsets = list.offsets();
+        for (i, sub_series) in list.into_iter().enumerate() {
+            let start_offset = list_offsets.get(i).unwrap();
+            if let Some(sub_series) = sub_series {
+                let probe_table = sub_series.build_probe_table_without_nulls()?;
+                let mut indices: Vec<_> = probe_table.keys().map(|k| k.idx).collect();
+                let unique_count = indices.len();
+                indices.sort_unstable();
```

Comment:
I wonder if we could use an IndexMap instead of a HashMap in `build_probe_table_without_nulls` to just maintain order instead of having to sort again. Feel free to just leave it as a todo